### PR TITLE
Lets people put hats on cyborgs

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
:cl: Kor
add: You can put a variety of hats on cyborgs using help intent (the engiborg can't wear hats though, as it is shaped too oddly. Sorry!)
add: The complete list of currently equippable hats is as follows: Cakehat, Captains Hat, Centcomm Hat, Witch Hunter Hat, HoS Cap, HoP Cap, Sombrero, Wizard Hat, Nurse Hat.
/:cl:

Reasoning: It's cute harmless customization that people have been asking for, it will be fun to put a cake hat on a medborg.

If anyone thinks of any other good hats (not helmets) that will fit will on borg heads, let me know.

Engiborg can't wear hats because his head is in a weird spot.

http://i.imgur.com/txjC1Io.png

### CODE REVIEW

**Comment by ChangelingRain** [comment]
Serviceborgs should be able to have hats, you just need to exclude hats from the tophat borg and properly set hat height for the kent borg.

**Comment by Niknakflak** [comment]
why is it restricted to certain hats if its coded in a way that's general?

**Comment by KorPhaeron** [comment]
Because a space helmet wouldn't look right halfway merged into a cyborgs head

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 1]
```diff
@@ -76,6 +77,12 @@
 
 	var/list/upgrades = list()
 
+	var/obj/item/hat
+	var/hat_offset = 3
```
Comment (ChangelingRain) [inline code/modules/mob/living/silicon/robot/robot.dm]: this default should be -3, right?

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 2]
```diff
@@ -937,6 +949,8 @@
 	speed = 0
 	ionpulse = FALSE
 
+	hat_offset = -3
```
Comment (KorPhaeron) [inline code/modules/mob/living/silicon/robot/robot.dm]: This is for setting back to no module/default

Comment (ChangelingRain) [inline code/modules/mob/living/silicon/robot/robot.dm]: The base module should handle that, though.

**Comment by Anonmare** [comment]
The beret and Sec beret come to mind. The Wizard and Witch hats as well. 
Engineer, CE and fire hard hats, as well as the orange version whatever it's called. 

Nurse hat as well, we need cute nurse mediborgs.

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 3]
```diff
@@ -339,6 +350,7 @@
 	moduleselect_icon = "service"
 	special_light_key = "service"
 	feedback_key = "cyborg_service"
+	hat_offset = INFINITY // No hats
```
Comment (ChangelingRain) [inline code/modules/mob/living/silicon/robot/robot_modules.dm]: Set this to 0, and in the `be_transformed_to()` for this module set it to INFINITY for Tophat and 3 for Kent.

**Comment by Cebasiz** [comment]
kor buddy

can we get a sprite for medals pinned to cyborg chassis please

thanks

**Comment by Iamgoofball** [comment]
@KorPhaeron this is where you apply blacklists instead of whitelists

**Comment by KorPhaeron** [comment]
Blacklists aren't maintainable

**Comment by Iamgoofball** [comment]
Whitelists aren't fun either

**Comment by RandomMarine** [comment]
How about a CYBORG_WEARABLE flag on hats instead?

**Comment by optimumtact** [comment]
Unmodular code that isn't worth it currently imo

Update the same dog fashion datum that ian uses for his hats - maybe rename (alternate_wearers or something similar) to be a bit cleaner, then despatch the borg related functionality

That way we justify the fashion datum more and the code for alternate hats isn't scattered through borg code, but centralised into a datum module.

**Comment by KorPhaeron** [comment]
I can look at it more, but fashions seemed more determined by which item is being used (new sprites per item), where as the look here is entirely determined by which module the borg has (same sprite, different position)

I'd either need to do istype for the borg module in the datum (and in the case of service borgs, a switch based on icon state) to figure out the pixel offset or else leave the offset stuff where it is now, which changes nothing so far as scattered code goes.

I might be misunderstanding what you're suggesting though.

**Comment by jughu** [comment]
Cueball head anyone?

**Comment by optimumtact** [comment]
BLOAT

**Comment by FantasticFwoosh** [comment]
Why not just give service borgs a chameleon hat module?

Now all we need is a accordion instrument, and then exclaim to remove non-human's with your steely cold eyes.

**Comment by Cebasiz** [comment]
make the accordion semi-robust melee weapon

give it attackverbs. include "removes"

**Comment by TehZombehz** [comment]
Is there anyway for a cyborg to remove a hat themselves or is this just another way to subject powerless cyborgs to the memes of the crew?

**Comment by KorPhaeron** [comment]
Cyborgs have no hands, how would they remove a hat?

**Comment by GunHog** [comment]
> Cyborgs have no hands, how would they remove a hat?

The same way Ian does: shaking.

**Comment by KorPhaeron** [comment]
If you think oranges dislikes the PR now I can't imagine a new borg button solely for removing hats will be well recieved

**Comment by GunHog** [comment]
You do not need a button, just use resist.

**Comment by coiax** [comment]
I am upset that engiborgs can't wear hats on the top of their casing, even if it's not where their "head" is

**Comment by xxalpha** [comment]
Borgs can remove it, yes?

**Comment by jughu** [comment]
Cant even wear a top hat

**Pull request merged**